ftw/linkchecker/report_generating.py

```diff
@@ -0,0 +1,74 @@
+import xlsxwriter
+from cell_format import BOLD, CENTER
+import time
+
+
+class ReportCreator(object):
+
+    def __init__(self, xlsx_file_path):
+        self.xlsx_file_path = xlsx_file_path
+        self.row = 0
+        self.workbook = xlsxwriter.Workbook(self.xlsx_file_path)
+        self.worksheet = self.workbook.add_worksheet()
+        self.autofitter_list_collector = []
+
+    def append_report_data(self, report_data, format=None):
+        format = format.get_workbook_format(self.workbook) if format else None
+
+        for row_list in (report_data):
+            col = 0
+            for value in row_list:
+                self.worksheet.write(self.row, col, value, format)
+                col += 1
+            self.row += 1
+        self.autofitter_list_collector.append(report_data)
+
+    def add_general_autofilter(self):
+        self.worksheet.autofilter(0, 0, self.row, len(labels[0]) - 1)
+
+    def cell_width_autofitter(self):
```

Comment:
please rename the vars used in this method. `i` and `x` are not too helpful to understand the code.

Reply:
I tried to make it better readable and also separated some code into another function.